src/models/Capability.js

```diff
@@ -2,14 +2,17 @@ import DB from "../lib/DB";
 import InsufficientPermissionsError from "../errors/InsufficientPermissionsError";
 
 export default class Capability {
-  static async getCapabilitiesByLoggedInUser(session_token) {
+  static async getCapabilitiesByLoggedInUser(session_token, params1, params2, params3) {
     const db = new DB();
 
     const sql = `SELECT DISTINCT capabilities.name from people_groups 
     LEFT JOIN groups ON groups.id = people_groups.group_id
     LEFT JOIN group_capabilities ON group_capabilities.group_id = groups.id
     LEFT JOIN capabilities ON capabilities.id = group_capabilities.capabilities_id
-    WHERE people_groups.people_id IN (select people_id from sessions where session = :session_token) AND capabilities.name IS NOT NULL`;
+    WHERE people_groups.people_id IN (select people_id from sessions where session = :session_token)
+    AND params1 ${params1 ? `= ${params1}` : 'IS NULL'}
+    AND params2 ${params2 ? `= ${params2}` : 'IS NULL'}
+    AND params3 ${params3 ? `= ${params3}` : 'IS NULL'}` 
```

Comment:
Hmmm, I'm quite uncomfortable that we are not putting the params inside the executeStatement parameters. However, I see your problem of putting it to null when needed. I'll look for a more elegant solution, but this has a security vulnerability.

Reply:
Can you try CASE inside WHERE? Caveat: I'm not sure if it will work.

https://stackoverflow.com/questions/14614573/using-case-in-the-where-clause

Reply:
Lucky guess
```suggestion
    AND CASE WHEN :params1 = "NULL" THEN  params1 IS NULL ELSE params1 = :params1 END
    AND params2 ${params2 ? `= ${params2}` : 'IS NULL'}
    AND params3 ${params3 ? `= ${params3}` : 'IS NULL'}` 
    
    const rawCapabilites = await db.executeStatement(sql, [
      { name: "session_token", value: { stringValue: session_token } },
       { name: "params1", value: { stringValue: params1 ?? "NULL" } },
    ]);   
```